test/unit/app/models/edition/lead_image_test.rb

```diff
@@ -2,41 +2,41 @@
 
 class Edition::LeadImageTest < ActiveSupport::TestCase
   test "should use placeholder image if none had been uploaded" do
-    model = stub("Target", images: [], lead_organisations: [], organisations: []).extend(Edition::LeadImage)
+    model = stub("Target", lead_image: nil, lead_organisations: [], organisations: []).extend(Edition::LeadImage)
```

Comment:
I know you didn't write the original test, but do you know why this test uses stub everywhere instead of just building a model? I definitely think we should try to avoid that in favour of using factories with build stubbed if at all possible. Might be worth a separate PR for a refactor.

Reply:
Maybe I'm too stub-averse, but I sort of feel the same about the stubbing done here, too:

https://github.com/alphagov/whitehall/blob/0005c564cbef8e74760c6c948eaf5a0449cdffbd/test/unit/app/models/edition/custom_lead_image_test.rb#L36-L37

It feels a bit brittle. What if we chain something else on to that Active Record call? e.g. `where()`

But presumably this is the 'cost' of using `build_stubbed` rather than `create` in these sorts of scenarios – e.g. when querying associations via the Active Record interface, that would usually expect data to have been persisted.

Reply:
yea unfortunately build stubbed isn't smart enough to handle active chaining active record queries. Generally if it's just one call like this i stub it out and any more i just use create. I just know we're trying to reduce our db hits.

Happy to use a create instead if thats the consensus.

Reply:
Regarding your first question, no i'm not actually sure. Happy to go back and refactor it to just use a case study or another model, but maybe in a follow up PR?

Reply:
It definitely feels like we're fighting ActiveRecord a bit trying to stub database reads. I do see ActiveRecord as sacrificing testability for convenience to a large degree, we're so coupled to the database it's very hard to test our logic without it. I'd probably take the hit on the create for Ollie's example in favour of a simpler test, personally

Reply:
okay have updated